dalm/datasets/reading_comprehension_generation/synthetic_based.py

```diff
@@ -0,0 +1,187 @@
+    "of sleep, some types of procedural memories were suggested not to rely on the hippocampus and involve REM phase of the sleep. "
+    "This inspired models where internal representations (memories) created by previous learning are spontaneously replayed during sleep-like periods in the network "
+    "itself (i.e. without help of secondary network performed by generative replay approaches mentioned above).\n"
+    "Question: [type: true/false] Is the following sentence true?  all types of procedural memories rely on the hippocampus\n"
+    "Answer: False. The text clearly states there are some types of procedural memories not reliant on the hippocampus\n--------\n"
+    "Question [type: complete-the-sentence] Complete the following sentence:  The insights into ____ in human and animal brain led to other _____ approaches\n"
+    "Answer: The insights into the mechanisms of memory consolidation during the sleep processes in human and animal brain led to other biologically inspired approaches\n------\n"
+    "Question [type 3 domain-keywords] Make a sentence with the following keywords 'hippocampo-neocortical', 'declarative' and 'NREM'\n"
+    "Answer: declarative memories are in the classical picture consolidated by hippocampo-neocortical dialog during NREM phase of sleep\n-------\n"
+    "Question [type: normal q&a] Some types of procedural memories were suggested not to rely on the hippocampus and involve REM phase of the sleep. What did this go on to inspire?\n"
+    "Answer This inspired models where internal representations (memories) created by previous learning are spontaneously replayed during sleep-like periods in the network itself [END OF EXAMPLE]\n\n "
+    "Similar to the above, could you craft 4 different reading comprehension tasks (make sure your output is a list of question answer pairs and each question is labelled QUESTION and answer is labelled ANSWER "
+    "and there is one question and answer per task) based solely and completely focused on the following TEXT: "
+)
+
+
+def gen_prompt(text: str) -> Dict[str, Any]:
+    prompt = PROMPT + text
+
+    return [
+        {
+            "role": "system",
+            "content": (
+                "You are a helpful and meticulous instruction following question and answer making chatbot. "
+                "Please refrain from acknowledgments, additions or niceties of any sort"
+            ),
+        },
+        {"role": "user", "content": prompt},
+    ]
+
+
+def generate_synthetic_data(model_pipeline: Pipeline, text: str, generation_params: Dict[str, Any]) -> str:
+    prompt = gen_prompt(text)
+    prompt = model_pipeline.tokenizer.apply_chat_template(prompt, tokenize=False, add_generation_prompt=True)
+    outputs = model_pipeline(prompt, **generation_params)
+
+    return outputs[0]["generated_text"]
+
+
+def generate_synthetic_dataset(
+    model_name: str,
+    input_directory: str,
+    processed_files: [str],
+    chunk: bool,
+    context_length: int,
```

Comment:
It would be helpful to callers of this API if there was a sensible default.  For example, the `default=4192` below could be pulled into a constant and used as a default in both places: in the function arg and the CLI default arg value.

Also any reason not to just name it `chunk_size`?  `context_length` implies that it's related to the model context window size, but it seems to be more related to how big to make the chunks for the dataset processing.

Reply:
@tleyden  I like to stay with the context_length since, then users will het a clear idea about the input to the model.